tests/OptionalMemorySafetyTest.cpp

```diff
@@ -56,6 +56,54 @@ static int s_OptionalMoveSafety(struct aws_allocator *allocator, void *ctx)
 
 AWS_TEST_CASE(OptionalMoveSafety, s_OptionalMoveSafety)
 
+class EmplaceTester
+{
+  public:
+    int a;
+    static size_t ctorCallCount;
+    static size_t dtorCallCount;
+    EmplaceTester(int val) : a(val) { ctorCallCount += 1; }
+    ~EmplaceTester()
+    {
+        a = -1337;
+        dtorCallCount += 1;
+    }
+    EmplaceTester(const EmplaceTester &) = delete;
+    EmplaceTester(EmplaceTester &&) = delete;
+};
+size_t EmplaceTester::ctorCallCount = 0;
+size_t EmplaceTester::dtorCallCount = 0;
+
+static int s_OptionalEmplace(struct aws_allocator *allocator, void *ctx)
+{
+    (void)ctx;
+    {
+        Aws::Crt::ApiHandle apiHandle(allocator);
+
+        Aws::Crt::Optional<Aws::Crt::String> str1{Aws::Crt::Optional<Aws::Crt::String>::in_place, s_test_str};
```

Comment:
[debatable] Maybe worth declaring `in_place` outside the class (like it is in std)? That would make using it shorter to type. We could add an `include/aws/crt/Utility.h` file so both Optional.h and Variant.h could use it:
```suggestion
        Aws::Crt::Optional<Aws::Crt::String> str1{Aws::Crt::InPlace, s_test_str};
```

Or we could keep it in `Optional.h`, but keep it outside the `Optional<T>` class, like:
```suggestion
        Aws::Crt::Optional<Aws::Crt::String> str1{Aws::Crt::OptionalInPlace, s_test_str};
```

Reply:
Thank you for a suggested change, I've updated the PR to do this.
(I'll rebase Variant PR after merging this PR to use the Utility header)